From 154016acc0982349f4471c35282eea886c9cbe57 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Wed, 19 Jul 2017 23:05:26 -0400
Subject: [PATCH 7/7] ACPI, APEI: Fixup incorrect 16-bit access width firmware
 bug

This is a follow up to commit f712c71f7b2b ("ACPI, APEI: Fixup common
access width firmware bug") fix the following firmware bug:

[Firmware Bug]: APEI: Invalid bit width + offset in GAR [0xb2/16/0/1/1]

This is due to an 8-bit access width is specified for a 16-bit register,
Do bit_width check just like what the original commit have done.

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 drivers/acpi/apei/apei-base.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/acpi/apei/apei-base.c b/drivers/acpi/apei/apei-base.c
index da370e1..7e47bc6 100644
--- a/drivers/acpi/apei/apei-base.c
+++ b/drivers/acpi/apei/apei-base.c
@@ -610,6 +610,9 @@ static int apei_check_gar(struct acpi_generic_address *reg, u64 *paddr,
 	else if (bit_width == 64 && bit_offset == 0 && (*paddr & 0x07) == 0 &&
 	    *access_bit_width < 64)
 		*access_bit_width = 64;
+	else if (bit_width == 16 && bit_offset == 0 && (*paddr & 0x01) == 0 &&
+	    *access_bit_width < 16)
+		*access_bit_width = 16;
 
 	if ((bit_width + bit_offset) > *access_bit_width) {
 		pr_warning(FW_BUG APEI_PFX
-- 
1.7.5.4

